lib/units/device/plugins/install.js

```
@@ -75,19 +79,52 @@ export default syrup.serial()
                 const stats = await fs.stat(destination.path)
                 log.info(`Downloaded file. Size: ${stats.size}`)
 
-
                 let target = '/data/local/tmp/install_app.apk'
-                await adb.getDevice(options.serial).shell(['rm', '-f', '/data/local/tmp/install_app.apk'])
+
+                try {
+                    log.info('Removing existing file if present')
+                    await adb.getDevice(options.serial).shell(['rm', '-f', target])
+                    log.info('Cleanup completed')
+                }
+                catch (error) {
+                    log.warn('Cleanup failed:', error instanceof Error ? error.message : String(error))
+                }
+
                 log.info('Started pushing apk')
-                const tr = await adb.getDevice(options.serial).push(destination.path, target, 0o755)
-                tr.on('error', log.error)
-                await tr.waitForEnd()
-                await destination.cleanup()
-                log.info('Installed app')
-                const apkstats = await adb.getDevice(options.serial).stat(target)
-                log.info(`File stats: ${JSON.stringify(apkstats)}`)
-                sendProgress('pushing_app', 50)
-                return target
+
+                const transfer = await adb.getDevice(options.serial).push(destination.path, target, 0o755)
+                transfer.on('progress', (stats) => {
+                    log.info(`Push progress: ${stats.bytesTransferred} bytes transferred`)
+                })
+                transfer.on('error', (error) => {
+                    log.error('Push transfer error:', error)
+                })
+
+                await transfer.waitForEnd()
```

This should throw an error if an error occured